pkgs/native_assets_cli/lib/src/architecture.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
```

Comment:
Is this a copy of the file from FFI? (Or is there no such file in FFI, and I've just dreamt it up)?

If so, don't copy unless they need to differ. Have a shared source instead.

Reply:
There doesn't seem to be a `class Architecture` in dart-lang/sdk (except in the test runner, but that one doesn't count). So I don't think it's a copy. The SDK does have `dart:ffi` `Abi` which is a combination of operating system and architecture.